node/fxutil/options.go

```diff
@@ -134,7 +134,7 @@ func OverrideSupply(vals ...interface{}) Option {
 				refVal, tp = refVal.Elem(), tp.Elem()
 			}
 
-			if !refVal.IsValid() {
+			if !refVal.IsValid() || refVal.IsZero() {
```

Comment:
Why the additional check? Seems orthogonal to the PR.

Reply:
I extracted it into a separate commit which explains the purpose here.

This makes overriding ignore zero values for built-in non-struct types.